src/walletSdk/Utils/soroban.ts

```diff
@@ -0,0 +1,84 @@
+import { Operation, StrKey, scValToNative, xdr } from "@stellar/stellar-sdk";
+
+import {
+  ArgsForTokenInvocation,
+  SorobanTokenInterface,
+  TokenInvocationArgs,
+} from "../Types";
+
+export const getArgsForTokenInvocation = (
+  fnName: string,
+  args: xdr.ScVal[],
+): ArgsForTokenInvocation => {
+  let amount: bigint | number;
+  let from = "";
+  let to = "";
+
+  switch (fnName) {
+    case SorobanTokenInterface.transfer:
+      from = StrKey.encodeEd25519PublicKey(
+        args[0].address().accountId().ed25519(),
+      );
+      to = StrKey.encodeEd25519PublicKey(
+        args[1].address().accountId().ed25519(),
+      );
+      amount = scValToNative(args[2]);
+      break;
+    case SorobanTokenInterface.mint:
+      to = StrKey.encodeEd25519PublicKey(
+        args[0].address().accountId().ed25519(),
+      );
+      amount = scValToNative(args[1]);
+      break;
+    default:
+      amount = BigInt(0);
```

Comment:
Should we throw here? It looks like this branch is inaccessible (we check that fnName is either transfer or mint before calling this function)

Reply:
@Ifropc good catch! I'm in favor of `throwing` an error here, and also throwing in all places that are returning `null` so we make it clear to devs where is the flaw. What do you guys think?

@piyalbasu do we have a strong reason for returning `null` in all places [here](https://github.com/stellar/freighter/blob/master/extension/src/popup/helpers/soroban.ts#L143) including the `try catch` blocks instead of throwing?

Reply:
So our reasoning for returning values was to guard against changing xdr spec's. We basically wanted to be able to feed this methods different invocations and not have it break if a specific underlying key was unexpectedly named differently.

I think it's probably a better and clearer API to throw errors in a case where a user did not pass an appropriate Transfer or Mint invocation. We'll just need to let devs know that these methods expect a valid Transfer or Mint invocation, so they will need to check that before using these.

Reply:
@piyalbasu thanks for the follow-up!

Just to clarify, are you suggesting we only throw on the code block below?
```
  if (
    ![SorobanTokenInterface.transfer, SorobanTokenInterface.mint].includes(
      fnName,
    )
  ) {
    throw new Error("Please use a valid Transfer or Mint invocation.");
  }
```

Reply:
Addressed this through a jsdoc comment here on the new PR: https://github.com/stellar/typescript-wallet-sdk/pull/119/files#diff-ebb0d904ab001286bca547ac642fe866bae6ffe07b009f97a6d8ad2d5aa5573eR40-R49